src/main/java/org/terasology/ligthandshadow/barrier/MagicDomeSystem.java

```diff
@@ -47,9 +51,16 @@ public void activateBarriers(ActivateBarrierEvent event, EntityRef entity) {
     }
 
     @ReceiveEvent
-    public void deactivateBarriers(DeactivateBarrierEvent event, EntityRef entity) {
-        redBarrier.destroy();
-        blackBarrier.destroy();
+    public void delayedDeactivateBarriers(DelayedDeactivateBarrierEvent event, EntityRef entity) {
+        delayManager.addDelayedAction(entity, "deactivate", 30000);
```

Comment:
A) I think the time after which the barriers should be destroyed should be defined by the event.

Reply:
B) These delayed action ids are global, and "deactivate" is pretty generic. I'd recommend to define a constant with a name resembling our typical asset urn schema, e.g., 

```java
private static final String DEACTIVATE_BARRIERS_ACTION = "LightAndShadow:deactivateBarriers";
```

This constant should be used both here and in the event handler for the delayed action